CesiumVectorData/include/CesiumVectorData/GeoJsonDocument.h

```diff
@@ -0,0 +1,140 @@
+#pragma once
+
+#include "CesiumGeospatial/Cartographic.h"
+#include "CesiumGeospatial/Ellipsoid.h"
+
+#include <CesiumAsync/AsyncSystem.h>
+#include <CesiumAsync/Future.h>
+#include <CesiumAsync/IAssetAccessor.h>
+#include <CesiumUtility/IntrusivePointer.h>
+#include <CesiumUtility/JsonValue.h>
+#include <CesiumUtility/ReferenceCounted.h>
+#include <CesiumUtility/Result.h>
+#include <CesiumVectorData/GeoJsonObject.h>
+#include <CesiumVectorData/Library.h>
+
+#include <memory>
+#include <span>
+
+namespace CesiumVectorData {
+
+/**
+ * @brief Attribution that must be included with a vector document.
+ */
+struct VectorDocumentAttribution {
+  /** @brief The HTML string containing attribution information to show. */
+  std::string html;
+  /**
+   * @brief If true, the attribution must be shown on screen. If false, it can
+   * be included in a popover instead.
+   */
+  bool showOnScreen;
+};
+
+/**
+ * @brief A vector document parsed from GeoJSON.
+ *
+ * The document is represented as a hierarchy of \ref GeoJsonObject values
+ * starting with the root object.
+ */
+class CESIUMVECTORDATA_API GeoJsonDocument
+    : public CesiumUtility::ReferenceCountedThreadSafe<GeoJsonDocument> {
+public:
+  /**
+   * @brief Attempts to parse a \ref GeoJsonDocument from the provided GeoJSON.
+   *
+   * @param bytes The GeoJSON data to parse.
+   * @param attributions Any attributions to attach to the document.
+   * @param ellipsoid The ellipsoid to use for this data.
+   * @returns A \ref CesiumUtility::Result containing the parsed
+   * \ref GeoJsonDocument or any errors and warnings that came up while parsing.
+   */
+  static CesiumUtility::Result<CesiumUtility::IntrusivePointer<GeoJsonDocument>>
+  fromGeoJson(
+      const std::span<const std::byte>& bytes,
+      std::vector<VectorDocumentAttribution>&& attributions = {},
+      const CesiumGeospatial::Ellipsoid& ellipsoid CESIUM_DEFAULT_ELLIPSOID);
+
+  /**
+   * @brief Attempts to parse a \ref GeoJsonDocument from the provided JSON
+   * document.
+   *
+   * @param document The GeoJSON JSON document.
+   * @param attributions Any attributions to attach to the document.
+   * @param ellipsoid The ellipsoid to use for this data.
+   * @returns A \ref CesiumUtility::Result containing the parsed
+   * \ref GeoJsonDocument or any errors and warnings that came up while parsing.
+   */
+  static CesiumUtility::Result<CesiumUtility::IntrusivePointer<GeoJsonDocument>>
+  fromGeoJson(
+      const rapidjson::Document& document,
+      std::vector<VectorDocumentAttribution>&& attributions = {},
+      const CesiumGeospatial::Ellipsoid& ellipsoid CESIUM_DEFAULT_ELLIPSOID);
+
+  /**
+   * @brief Attempts to load a \ref GeoJsonDocument from a Cesium ion asset.
+   *
+   * Currently only the GeoJSON format is supported.
```

Comment:
Redundant comment? 😆 

```suggestion
   * @brief Attempts to load a \ref GeoJsonDocument from a Cesium ion asset.
```